internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -388,6 +389,24 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 			}
 		}
 	}
+	patchHelper, err := patch.NewHelper(cluster, r.Client)
```

Comment:
Wondering if we can benefit of the deferred patch helper in the main reconcile method...

What if we move 

https://github.com/kubernetes-sigs/cluster-api/blob/282fa77eaaa6d1040dfc3f0d1e53aca21f0c6e25/internal/controllers/topology/cluster/cluster_controller.go#L168-L172

Down after the defer?
This will probably require also to make reconcileConditions no-op in case the cluster is deleted, or even better we should move the logic to set the condition while deleting from here to reconcileConditions

Opinions?

Reply:
👋 based on [this comment](https://github.com/kubernetes-sigs/cluster-api/blob/282fa77eaaa6d1040dfc3f0d1e53aca21f0c6e25/internal/controllers/topology/cluster/cluster_controller.go#L168):
> // In case the object is deleted, the managed topology stops to reconcile;
 
I assumed that the idea was to prevent [the reconcile method](https://github.com/kubernetes-sigs/cluster-api/blob/282fa77eaaa6d1040dfc3f0d1e53aca21f0c6e25/internal/controllers/topology/cluster/cluster_controller.go#L200-L201) from being called once the `DeletionTimestamp` is recognized.  

If the logic is moved to `reconcileConditions` then the normal reconciliation loop will run when the Cluster object has been deleted.  Is that okay?

Reply:
hey @fabriziopandini wanted to bump this and see if you have any feedback on my most recent comment

Reply:
I believe what @fabriziopandini Is suggesting is to move the entire if block 
```golang
        // In case the object is deleted, the managed topology stops to reconcile;
	// (the other controllers will take care of deletion).
	if !cluster.ObjectMeta.DeletionTimestamp.IsZero() {
		return r.reconcileDelete(ctx, cluster)
	}
```
below the defer block but still keep it above the call to `reconcile()` so that if the the cluster is deleted it only calls `reconcileDelete` and not `reconcile`. 
The defer block will be triggered after `reconcileDelete` returns and inside the defer block the `reconcileConditions` should handle setting the `TopologyReconciled` condition depending on if the cluster is deleted or not.

Reply:
sounds good, thanks @ykakarap.  i just pushed up these changes -- let me know if you have additional feedback